### PR TITLE
fix #1794, split dispatchers of Task and Netty

### LINKED ISSUE #1794: split dispatchers of Task and Netty

### PR DESCRIPTION
**Pull request by manuzhang**
Open for early review so do not merge. I still need to do more verification and add docs.

### CODE REVIEW

File: conf/gear.conf [hunk 1]
```diff
@@ -63,7 +63,11 @@ gearpump {
   ### If you don't know what this is about, don't change it
   ###########################
 
-  task-dispatcher {
+  task-dispatcher = "shared-thread-pool"
```
Comment (clockfly) [inline conf/gear.conf]: The value should be full config key, and better use -dispatcher as suffix.

For example, For Akka-remoting, it use the dispatcher like this:
```
akka.remote.use-dispatcher = "akka.remote.default-remote-dispatcher"
```

**Comment by clockfly** [comment]
The idea looks good!

**Comment by manuzhang** [comment]
@clockfly updated and fixed docs as well

**Comment by clockfly** [comment]
Looks good! +1

Can you also update the user doc about the common pitfall to avoid in FAQ and storm section

**Comment by manuzhang** [comment]
updated

File: docs/dev-storm.md [hunk 2]
```diff
@@ -58,6 +58,11 @@ At runtime, spouts and bolts are running inside `StormProducer` tasks and `Storm
 
 Storm ackers are dropped since Gearpump has a different mechanism of message tracking and flow control. 
 
+### Task execution
+
+Each Storm task is executed by a dedicated thread while all Gearpump tasks of an executor share a thread pool. Generally, we can achieve better performance with a shared thread pool. It's possible, however, some tasks block and take up all the threads. In that case, we can 
+fall back to the Storm way by setting `task-dispatcher` to `"gaerpump.single-thread-dispatcher"` in `gear.conf`.
```
Comment (clockfly) [inline docs/dev-storm.md]: by setting "gearpump.dispatcher"

File: docs/faq.md [hunk 3]
```diff
@@ -6,26 +6,32 @@ description: Gearpump Frequently Asked Questions
...
-1. It is more easy to conduct Unit test
-2. We have custom logic and messages to ensure the data consistency, like flow control, like message loss detection.
-3. As the Gearpump interface evolves rapidly. for now, we want to conservative in exposing more powerful functions so that we doesn't tie our hands for future refactory, it let us feel safe.
+1. It is easier to conduct Unit test
+2. We have custom logic and messages to ensure the data consistency, like flow control and message loss detection.
+3. As the Gearpump interface evolves rapidly, for now, we want to be conservative in exposing more powerful functions so that we don't tie our hands for future refactoring. It lets us feel safe.
+
+##### Why does my task has extremely high message latency (e.g. 10 seconds) ?
+
+Please check whether you are doing blocking jobs (e.g. sleep, IO) in your task. By default, all tasks in an executor share a thread pool. The blocking tasks could use up all the threads while other tasks don't get a chance to run. In that case, you can set `task-dispatcher` to `"gearpump.single-thread-dispatcher"` in `gear.conf` such that a unique thread is dedicated to each task.
```
Comment (clockfly) [inline docs/faq.md]: "gearpump.task-dispatcher"

**Comment by clockfly** [comment]
+1

**Pull request merged**